document/core/exec/instructions.rst

```diff
@@ -651,27 +651,27 @@ Table Instructions
 
 11. Pop the value :math:`\I32.\CONST~i` from the stack.
 
-12. If :math:`n` is :math:`0`, then:
+12. If :math:`i + n` is larger than the length of :math:`\X{tab}.\TIELEM`, then:
 
-    a. If :math:`i` is larger than the length of :math:`\X{tab}.\TIELEM`, then:
+    a. Trap.
 
-       i. Trap.
+12. If :math:`n` is :math:`0`, then:
 
-12. Else:
+    a. Return.
 
-    a. Push the value :math:`\I32.CONST~i` to the stack.
+13. Push the value :math:`\I32.CONST~i` to the stack.
 
-    b. Push the value :math:`\val` to the stack.
+14. Push the value :math:`\val` to the stack.
 
-    c. Execute the instruction :math:`\TABLESET~x`.
+15. Execute the instruction :math:`\TABLESET~x`.
 
-    d. Push the value :math:`\I32.CONST~(i+1)` to the stack.
+16. Push the value :math:`\I32.CONST~(i+1)` to the stack.
 
-    e. Push the value :math:`\val` to the stack.
+17. Push the value :math:`\val` to the stack.
 
-    f. Push the value :math:`\I32.CONST~(n-1)` to the stack.
+18. Push the value :math:`\I32.CONST~(n-1)` to the stack.
 
-    c. Execute the instruction :math:`\TABLEFILL~x`.
+19. Execute the instruction :math:`\TABLEFILL~x`.
 
 .. math::
    \begin{array}{l}
```

Comment:
The formal notation needs to be fixed too -- this will still partially fill.

Reply:
Perhaps someone else could fix that part? I'm not super familiar with latex / our conventions for the formal notation.